### ISSUE TITLE
kedro viz > 6.5.0 times out when working with Spark

### ISSUE THREAD

**Issue by kennyb-qb** [kedro-org/kedro-viz#1768]
## Description
tried to use latest kedro-viz (which uses command `kedro viz run`) with kedro-datasets (latest version) and kedro version 0.18.14.

## Context
cannot use `kedro viz run`. 
when I remove kedro-datasets dependency and downgrade to kedro-viz<6 then I can get kedro viz running.

## Steps to Reproduce
1. AWS workspace windows 10
2. pip install kedro~=0.18.14 kedro-datasets kedro-viz
3. have kedro codebase with SparkDataset defined in catalog and pipeline
4. kedro viz run

## Expected Result
kedro viz should come up on local

## Actual Result
viz check timeout error

```
Traceback (most recent call last):
  File "D:\Users\buik4\AppData\Local\miniconda3\envs\py39\lib\site-packages\kedro_viz\launchers\cli.py", line 187, in run
    _wait_for(func=_check_viz_up, host=host, port=port)
  File "D:\Users\buik4\AppData\Local\miniconda3\envs\py39\lib\site-packages\kedro_viz\launchers\utils.py", line 58, in _wait_for
    raise WaitForException(
kedro_viz.launchers.utils.WaitForException: func: <function _check_viz_up at 0x000001BDD5E46310>, didn't return True within specified timeout: 60
kedro.framework.cli.utils.KedroCliError: func: <function _check_viz_up at 0x000001BDD5E46310>, didn't return True within specified timeout: 60
Run with --verbose to see the full exception
Error: func: <function _check_viz_up at 0x000001BDD5E46310>, didn't return True within specified timeout: 60
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:
* Kedro version used (`pip show kedro` or `kedro -V`):  0.18.14
* Kedro plugin and kedro plugin version used (`pip show kedro-airflow`): kedro-viz 7.1.0, 7.0.0, 6.7.0 | kedro-datasets 2.0.0, 1.0.0
* Python version used (`python -V`): 3.9.18 (miniconda)
* Operating system and version: Windows 10 (AWS workspace)

**Comment by astrojuanlu** [comment]
Hi @kennyb-qb , thanks for reporting and sorry you're having a bumpy ride. Could you share a `pip freeze` of the environment to better understand your current dependencies?

Also, could you launch `kedro viz run` with `--verbose` to see the full traceback?

**Comment by kennyb-qb** [comment]
`--verbose` option does not exist.
<img width="680" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/101136683/408e7f35-31ff-48bd-a17f-a30a8ac19833">

**Comment by kennyb-qb** [comment]
```
>pip freeze
aiobotocore==2.11.2
aiofiles==22.1.0
aiohttp==3.9.3
aioitertools==0.11.0
aiosignal==1.3.1
aiosqlite==0.20.0
antlr4-python3-runtime==4.9.3
anyconfig==0.14.0
anyio==3.7.1
archspec @ file:///croot/archspec_1697725767277/work
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asn1crypto==1.5.1
asttokens==2.4.1
async-timeout==4.0.3
attrs @ file:///C:/b/abs_35n0jusce8/croot/attrs_1695717880170/work
awscli==1.32.50
Babel==2.14.0
beautifulsoup4 @ file:///C:/b/abs_0agyz1wsr4/croot/beautifulsoup4-split_1681493048687/work
binaryornot==0.4.4
black==22.12.0
bleach==6.1.0
boltons @ file:///C:/b/abs_707eo7c09t/croot/boltons_1677628723117/work
boto3==1.34.50
botocore==1.34.34
Brotli @ file:///C:/Windows/Temp/abs_63l7912z0e/croots/recipe/brotli-split_1659616056886/work
build==1.0.3
cachetools==5.3.3
certifi @ file:///C:/b/abs_35d7n66oz9/croot/certifi_1707229248467/work/certifi
cffi @ file:///C:/b/abs_924gv1kxzj/croot/cffi_1700254355075/work
chardet @ file:///C:/ci/chardet_1607706937985/work
charset-normalizer @ file:///tmp/build/80754af9/charset-normalizer_1630003229654/work
click @ file:///C:/b/abs_f9ihnt72pu/croot/click_1698129847492/work
colorama==0.4.4
comm==0.2.1
conda @ file:///C:/b/abs_89vd8hj61u/croot/conda_1708369170790/work
conda-build @ file:///C:/b/abs_3ed9gavxgz/croot/conda-build_1708025907525/work
conda-libmamba-solver @ file:///croot/conda-libmamba-solver_1706733287605/work/src
conda-package-handling @ file:///C:/b/abs_b9wp3lr1gn/croot/conda-package-handling_1691008700066/work
conda_index @ file:///croot/conda-index_1706633791028/work
conda_package_streaming @ file:///C:/b/abs_6c28n38aaj/croot/conda-package-streaming_1690988019210/work
cookiecutter==2.6.0
coverage==7.4.3
debugpy==1.8.1
decorator==5.1.1
defusedxml==0.7.1
distro @ file:///C:/b/abs_a3uni_yez3/croot/distro_1701455052240/work
docopt==0.6.2
docutils==0.16
dynaconf==3.2.4
entrypoints==0.4
exceptiongroup==1.2.0
executing==2.0.1
fastapi==0.110.0
fastjsonschema==2.19.1
filelock @ file:///C:/b/abs_f2gie28u58/croot/filelock_1700591233643/work
fqdn==1.5.1
frozenlist==1.4.1
fsspec==2023.12.2
gitdb==4.0.11
GitPython==3.1.42
graphql-core==3.2.3
greenlet==3.0.3
h11==0.14.0
hdfs==2.7.3
httptools==0.6.1
idna @ file:///C:/b/abs_bdhbebrioa/croot/idna_1666125572046/work
importlib-metadata==6.11.0
importlib_resources==6.1.2
iniconfig==2.0.0
ipykernel==6.29.3
ipython==8.18.1
ipython-genutils==0.2.0
ipywidgets==8.1.2
isoduration==20.11.0
jedi==0.19.1
Jinja2 @ file:///C:/b/abs_f7x5a8op2h/croot/jinja2_1706733672594/work
jmespath==1.0.1
json5==0.9.17
jsonpatch @ file:///tmp/build/80754af9/jsonpatch_1615747632069/work
jsonpointer==2.1
jsonschema @ file:///C:/b/abs_d1c4sm8drk/croot/jsonschema_1699041668863/work
jsonschema-specifications @ file:///C:/b/abs_0brvm6vryw/croot/jsonschema-specifications_1699032417323/work
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.9.0
jupyter-ydoc==0.2.5
jupyter_client==7.4.9
jupyter_core==5.7.1
jupyter_server==2.12.5
jupyter_server_fileid==0.9.1
jupyter_server_terminals==0.5.2
jupyter_server_ydoc==0.8.0
jupyterlab==3.6.7
jupyterlab_pygments==0.3.0
jupyterlab_server==2.25.3
jupyterlab_widgets==3.0.10
kedro==0.18.14
kedro-datasets==2.0.0
kedro-telemetry==0.2.5
kedro-viz==7.1.0
lazy_loader==0.3
libarchive-c @ file:///tmp/build/80754af9/python-libarchive-c_1617780486945/work
libmambapy @ file:///C:/b/abs_2euls_1a38/croot/mamba-split_1704219444888/work/libmambapy
lxml==5.1.0
markdown-it-py==3.0.0
MarkupSafe @ file:///C:/b/abs_ecfdqh67b_/croot/markupsafe_1704206030535/work
matplotlib-inline==0.1.6
mdurl==0.1.2
menuinst @ file:///C:/b/abs_099kybla52/croot/menuinst_1706732987063/work
mistune==3.0.2
more-itertools @ file:///C:/b/abs_36p38zj5jx/croot/more-itertools_1700662194485/work
multidict==6.0.5
mypy-extensions==1.0.0
nbclassic==1.0.0
nbclient==0.9.0
nbconvert==7.16.1
nbformat==5.9.2
nest-asyncio==1.6.0
networkx==3.2.1
notebook==6.5.6
notebook_shim==0.2.4
numpy==1.26.4
omegaconf==2.3.0
orjson==3.9.15
overrides==7.7.0
packaging==23.2
pandas==2.2.1
pandocfilters==1.5.1
parse==1.20.1
parso==0.8.3
pathspec==0.12.1
patsy==0.5.6
pip-tools==7.4.0
pkginfo @ file:///C:/b/abs_d18srtr68x/croot/pkginfo_1679431192239/work
platformdirs==4.2.0
plotly==5.19.0
pluggy @ file:///C:/ci/pluggy_1648024580010/work
prometheus_client==0.20.0
prompt-toolkit==3.0.43
psutil @ file:///C:/Windows/Temp/abs_b2c2fd7f-9fd5-4756-95ea-8aed74d0039flsd9qufz/croots/recipe/psutil_1656431277748/work
pure-eval==0.2.2
py4j==0.10.9.7
pyarrow==15.0.0
pyasn1==0.5.1
pycosat @ file:///C:/b/abs_31zywn1be3/croot/pycosat_1696537126223/work
pycparser @ file:///tmp/build/80754af9/pycparser_1636541352034/work
pydantic==1.10.14
Pygments==2.17.2
pyproject_hooks==1.0.0
PySocks @ file:///C:/ci/pysocks_1605307512533/work
pyspark==3.5.1
pytest==7.4.4
pytest-cov==3.0.0
pytest-mock==1.13.0
python-dateutil==2.8.2
python-dotenv==1.0.1
python-json-logger==2.0.7
python-slugify==8.0.4
pytoolconfig==1.3.1
pytz @ file:///C:/b/abs_19q3ljkez4/croot/pytz_1695131651401/work
pywin32==306
pywinpty==2.0.13
PyYAML @ file:///C:/b/abs_782o3mbw7z/croot/pyyaml_1698096085010/work
pyzmq==24.0.1
qtconsole==5.5.1
QtPy==2.4.1
redshift-connector==2.1.0
referencing @ file:///C:/b/abs_09f4hj6adf/croot/referencing_1699012097448/work
requests @ file:///C:/b/abs_474vaa3x9e/croot/requests_1707355619957/work
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.7.0
rope==1.12.0
rpds-py @ file:///C:/b/abs_76j4g4la23/croot/rpds-py_1698947348047/work
rsa==4.7.2
ruamel.yaml @ file:///C:/b/abs_30ee5qbthd/croot/ruamel.yaml_1666304562000/work
ruamel.yaml.clib @ file:///C:/b/abs_aarblxbilo/croot/ruamel.yaml.clib_1666302270884/work
ruff==0.0.292
s3fs==2023.12.2
s3transfer==0.10.0
scipy==1.12.0
scramp==1.4.4
secure==0.3.0
Send2Trash==1.8.2
six==1.16.0
smmap==5.0.1
sniffio==1.3.1
soupsieve @ file:///C:/b/abs_bbsvy9t4pl/croot/soupsieve_1696347611357/work
SQLAlchemy==2.0.27
stack-data==0.6.3
starlette==0.36.3
statsmodels==0.14.1
strawberry-graphql==0.219.2
tenacity==8.2.3
terminado==0.18.0
text-unidecode==1.3
tinycss2==1.2.1
toml==0.10.2
tomli @ file:///C:/Windows/TEMP/abs_ac109f85-a7b3-4b4d-bcfd-52622eceddf0hy332ojo/croots/recipe/tomli_1657175513137/work
toposort==1.10
tornado==6.4
tqdm @ file:///C:/b/abs_f76j9hg7pv/croot/tqdm_1679561871187/work
traitlets==5.14.1
types-python-dateutil==2.8.19.20240106
typing_extensions==4.10.0
tzdata==2024.1
uri-template==1.3.0
urllib3==1.26.18
uvicorn==0.25.0
watchfiles==0.21.0
watchgod==0.8.2
wcwidth==0.2.13
webcolors==1.13
webencodings==0.5.1
websocket-client==1.7.0
websockets==12.0
widgetsnbextension==4.0.10
win-inet-pton @ file:///C:/ci/win_inet_pton_1605306162074/work
wrapt==1.16.0
y-py==0.6.2
yarl==1.9.4
ypy-websocket==0.8.4
zipp==3.17.0
zstandard==0.19.0
```

**Comment by rashidakanchwala** [comment]
@kennyb-qb @astrojuanlu  - we were not able to replicate this on our Macbooks so it feels like a Windows problem. We are doing a release coming Monday, so I will ask you to see if the new release fixing the issues. If not, we are going to look into this problem in our next sprint starting Monday.

**Comment by kennyb-qb** [comment]
@rashidakanchwala will `--verbose` option be available in the new release?
cannot show traceback without this.

**Comment by ravi-kumar-pilla** [comment]
Hi @kennyb-qb,

Thank you for your patience. I tried to replicate the issue on Amazon Workspaces with the below configuration -

1. **OS** - Windows Server 2022
2. **Launch Bundle** - Standard with Windows 10 (Server 2022 based) (WSP)
3. **Package versions** - `Python 3.9`, `Kedro 0.18.14,`` Kedro-datasets==2.0.0`, `Kedro-Viz (7.1.0, 7.0.0, 6.7.0)`

Initial smoke tests were conducted on `spaceflights starter project` which seems to work fine. I added a `spark.SparkDataset` to the catalog to see if **SparkDataset** is causing the issue. However, I could not replicate the issue. We would like to investigate more on the bug if this happens on Windows alone or due to any other factors. 

I would request you to test this with the above configuration with the starter project (`kedro new --starter=spaceflights`) in case you have some time. Also as @rashidakanchwala pointed out, this may get fixed automatically with the `Kedro-Viz 8.0.0` release. If not, we are happy to debug the issue further.

**Test screenshots** :

**Kedro-Viz - 7.1.0** 

![Screenshot 2024-02-29 at 5 58 06 PM](https://github.com/kedro-org/kedro-viz/assets/87735534/bead72bb-bf93-4705-a414-59cbf1434947)

![Screenshot 2024-02-29 at 5 47 52 PM](https://github.com/kedro-org/kedro-viz/assets/87735534/6e42a4b7-d918-4d82-85ca-c3ab2a5564cf)

**Kedro-Viz - 7.0.0** 

![Screenshot 2024-02-29 at 5 45 33 PM](https://github.com/kedro-org/kedro-viz/assets/87735534/b2ad691c-4002-42c6-83dd-bd82c92e9e8b)

![Screenshot 2024-02-29 at 5 45 57 PM](https://github.com/kedro-org/kedro-viz/assets/87735534/094f7a5b-e94b-4a26-a863-1fddf60b3800)

**Comment by ravi-kumar-pilla** [comment]
Hi @kennyb-qb,

I wanted to let you know that Kedro-Viz v8.0.0 is out and if you have time, please upgrade your Kedro-Viz version to the latest release and let me know if the issue still exists. 

Also, Kedro-Viz 8.0.0 is compatible with Kedro-datasets >= 2.1.0. Please checkout the [release notes](https://github.com/kedro-org/kedro-viz/releases/tag/v8.0.0) for more details.

Thank you

**Comment by kennyb-qb** [comment]
latest fix did not work
kedro-dataset 2.1.0
kedro-viz 8.0.1
kedro 0.18.14

also, I have SparkSession setup in my hooks. don't think the spaceflight tutorial would have that.
i've also installed spark similar to this: https://www.machinelearningplus.com/pyspark/install-pyspark-on-windows/
with these hadoop bins: https://github.com/cdarlint/winutils

**Comment by astrojuanlu** [comment]
I'm not fully up to date with the conversation but looking at #1803, I understand that there's no incompatibility here, but rather a timing issue with the initialization of Spark? @ravi-kumar-pilla do you mind updating the title of the issue? (I'd do it myself but again I'm not 100 % sure what's going on)

**Comment by ravi-kumar-pilla** [comment]
> I'm not fully up to date with the conversation but looking at #1803, I understand that there's no incompatibility here, but rather a timing issue with the initialization of Spark? @ravi-kumar-pilla do you mind updating the title of the issue? (I'd do it myself but again I'm not 100 % sure what's going on)

Sure @astrojuanlu . For some context, I spoke to @kennyb-qb and found out that the initialization of Spark takes close to 2min. Since we introduced a timeout for kedro-viz from v6.6.0, it times out. As a temporary fix we have - #1803 .
